host/xdc/history_event_test.go

```diff
@@ -0,0 +1,247 @@
+	cwfInitialFail := NewHistoryEvent(shared.EventTypeStartChildWorkflowExecutionFailed.String())
+	cwfStart := NewHistoryEvent(shared.EventTypeChildWorkflowExecutionStarted.String())
+	cwfCancel := NewHistoryEvent(shared.EventTypeChildWorkflowExecutionCanceled.String())
+	cwfComplete := NewHistoryEvent(shared.EventTypeChildWorkflowExecutionCompleted.String())
+	cwfFail := NewHistoryEvent(shared.EventTypeChildWorkflowExecutionFailed.String())
+	cwfTerminate := NewHistoryEvent(shared.EventTypeChildWorkflowExecutionTerminated.String())
+	cwfTimedOut := NewHistoryEvent(shared.EventTypeChildWorkflowExecutionTimedOut.String())
+
+	dtCompleteToCWFInitial := NewConnection(dtComplete, cwfInitial)
+	cwfInitialToFail := NewConnection(cwfInitial, cwfInitialFail)
+	cwfInitialToStart := NewConnection(cwfInitial, cwfStart)
+	cwfStartToCancel := NewConnection(cwfStart, cwfCancel)
+	cwfStartToFail := NewConnection(cwfStart, cwfFail)
+	cwfStartToComplete := NewConnection(cwfStart, cwfComplete)
+	cwfStartToTerminate := NewConnection(cwfStart, cwfTerminate)
+	cwfStartToTimedOut := NewConnection(cwfStart, cwfTimedOut)
+	cwfCancelToDTSchedule := NewConnection(cwfCancel, dtSchedule)
+	cwfCancelToDTSchedule.SetCondition(notPendingDecisionTask)
+	cwfFailToDTSchedule := NewConnection(cwfFail, dtSchedule)
+	cwfFailToDTSchedule.SetCondition(notPendingDecisionTask)
+	cwfCompleteToDTSchedule := NewConnection(cwfComplete, dtSchedule)
+	cwfCompleteToDTSchedule.SetCondition(notPendingDecisionTask)
+	cwfTerminateToDTSchedule := NewConnection(cwfTerminate, dtSchedule)
+	cwfTerminateToDTSchedule.SetCondition(notPendingDecisionTask)
+	cwfTimedOutToDTSchedule := NewConnection(cwfTimedOut, dtSchedule)
+	cwfTimedOutToDTSchedule.SetCondition(notPendingDecisionTask)
+	cwfInitialFailToDTSchedule := NewConnection(cwfInitialFail, dtSchedule)
+	cwfInitialFailToDTSchedule.SetCondition(notPendingDecisionTask)
+	cwfModel.AddEdge(dtCompleteToCWFInitial, cwfInitialToFail, cwfInitialToStart, cwfStartToCancel, cwfStartToFail,
+		cwfStartToComplete, cwfStartToTerminate, cwfStartToTimedOut, cwfCancelToDTSchedule, cwfFailToDTSchedule,
+		cwfCompleteToDTSchedule, cwfTerminateToDTSchedule, cwfTimedOutToDTSchedule, cwfInitialFailToDTSchedule)
+
+	ewfModel := NewHistoryEventModel()
+	ewfSignal := NewHistoryEvent(shared.EventTypeSignalExternalWorkflowExecutionInitiated.String())
+	ewfSignalFailed := NewHistoryEvent(shared.EventTypeSignalExternalWorkflowExecutionFailed.String())
+	ewfSignaled := NewHistoryEvent(shared.EventTypeExternalWorkflowExecutionSignaled.String())
+	ewfCancel := NewHistoryEvent(shared.EventTypeRequestCancelExternalWorkflowExecutionInitiated.String())
+	ewfCancelFail := NewHistoryEvent(shared.EventTypeRequestCancelExternalWorkflowExecutionFailed.String())
+	ewfCanceled := NewHistoryEvent(shared.EventTypeExternalWorkflowExecutionCancelRequested.String())
+
+	dtCompleteToEWFSignal := NewConnection(dtComplete, ewfSignal)
```

Comment:
try use variable names that is more clear, like decision timeout? i guess for this case

Reply:
Replace all the short names with readable names